ipv8/messaging/payload.py

```diff
@@ -160,29 +166,42 @@ def to_pack_list(self):
                 ('4SH', inet_aton(self.source_wan_address[0]), self.source_wan_address[1]),
                 ('4SH', inet_aton(self.lan_introduction_address[0]), self.lan_introduction_address[1]),
                 ('4SH', inet_aton(self.wan_introduction_address[0]), self.wan_introduction_address[1]),
-                ('bits', encoded_connection_type[0], encoded_connection_type[1], 0, 0, 0, 0, 0, 0),
+                ('bits', encoded_connection_type[0], encoded_connection_type[1], 0, self.supports_new_style,
+                 self.intro_supports_new_style, 0, 0, 0),
                 ('H', self.identifier),
                 ('raw', self.extra_bytes)]
         return data
 
     @classmethod
     def from_unpack_list(cls, destination_address, source_lan_address, source_wan_address,
                          introduction_lan_address, introduction_wan_address,
-                         connection_type_0, connection_type_1, dflag0, dflag1, dflag2, dflag3, dflag4, dflag5,
-                         identifier, extra_bytes):
-        args = [(inet_ntoa(destination_address[0]), destination_address[1]),
-                (inet_ntoa(source_lan_address[0]), source_lan_address[1]),
-                (inet_ntoa(source_wan_address[0]), source_wan_address[1]),
-                (inet_ntoa(introduction_lan_address[0]), introduction_lan_address[1]),
-                (inet_ntoa(introduction_wan_address[0]), introduction_wan_address[1]),
+                         connection_type_0, connection_type_1, dflag0, supports_new_style, intro_supports_new_style,
+                         dflag3, dflag4, dflag5, identifier, extra_bytes):
+        args = [UDPv4Address(inet_ntoa(destination_address[0]), destination_address[1]),
```

Comment:
For aesthetic reasons, you could modify the `IPv4` packer to return `UDPv4Address` tuples, and switch from `4SH` to `ipv4` in the payload.

Reply:
Done 👍